tools/crm_verify.c

```diff
@@ -148,28 +148,17 @@ main(int argc, char **argv)
     crm_info("=#=#=#=#= Getting XML =#=#=#=#=");
 
     if (options.use_live_cib) {
-        cib_conn = cib_new();
-        rc = cib_conn->cmds->signon(cib_conn, crm_system_name, cib_command);
-        rc = pcmk_legacy2rc(rc);
-    }
-
-    if (options.use_live_cib) {
-        if (rc == pcmk_rc_ok) {
-            int options = cib_scope_local | cib_sync_call;
+        crm_info("Reading XML from: live cluster");
+        rc = cib__signon_query(NULL, &cib_object);
 
-            crm_info("Reading XML from: live cluster");
-            rc = cib_conn->cmds->query(cib_conn, NULL, &cib_object, options);
-            rc = pcmk_legacy2rc(rc);
-        }
-
-        if (rc != pcmk_rc_ok) {
-            g_set_error(&error, PCMK__RC_ERROR, rc, "Live CIB query failed: %s", pcmk_rc_str(rc));
-            goto done;
-        }
-        if (cib_object == NULL) {
+        if (rc == pcmk_rc_no_input) {
             rc = ENOMSG;
             g_set_error(&error, PCMK__RC_ERROR, rc, "Live CIB query failed: empty result");
             goto done;
+
+        } else if (rc != pcmk_rc_ok) {
+            g_set_error(&error, PCMK__RC_ERROR, rc, "Live CIB query failed: %s", pcmk_rc_str(rc));
+            goto done;
         }
```

Comment:
Related to error handling, note the commit message for 2ef68d5185440b8509ea9f6ecbc854fc526bec0e and what has happened to all the GError usage there.

Reply:
I do think the functions should return standard error codes, and the mains should set GError from it if desired. But if we're setting GError differently for different codes, we're probably using the wrong codes -- pcmk_rc_str() should give something appropriate.

Reply:
The problem seems to be more that the simulate function could return (for instance) some errno value that could either be because writing the dot file failed, or because writing the XML file failed.  Without the GError, the caller won't know which the errno corresponds to.

Reply:
Right, that means the simulate function isn't returning sufficient codes